doc/manual/src/protocols/signature.md

```diff
@@ -0,0 +1,49 @@
+# Signature
+
+In Nix, signatures are used to trust store paths using asymmetric cryptography,
+in this instance: Curve25519-based cryptography.
+
+## Store path signature format
+
+Store path signatures are ED25519 signatures, there are two types of signatures:
+
+- `ValidPathInfo` will assemble a fingerprint in the form of
+  ```
+  1;<store path>;<NAR hash in base32 including its own type>;<NAR size in bytes>;<NAR references separated by ,>
+  ```
+  and sign this information with the ED25519 private key.
```

Comment:
I'm not sure what the exact signature scheme is, but if we intend to sign prehashed payload (and use the `NAR hash in base32` specified here), ed25519 requires it to be sha512.
(https://docs.rs/ed25519-dalek/latest/ed25519_dalek/trait.DigestSigner.html#impl-DigestSigner%3CD,+Signature%3E-for-Context%3C'_,+'_,+SigningKey%3E)